app/java/src/main/java/com/google/cloudclientapi/CensusController.java

```diff
@@ -16,28 +16,65 @@
 
 package com.google.cloudclientapi;
 
-
+import com.google.gson.Gson;
 import java.util.Arrays;
 import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
 @Controller
 public class CensusController {
+
+  private static final String PROCESSED_DATA_BUCKET = System.getenv().get("PROCESSED_DATA_BUCKET");
   private static final Logger logger = LoggerFactory.getLogger(CensusController.class);
 
   @GetMapping("/")
-  public String main(Model model) {
-    logger.info("Received request (TODO: data)");
+  public String main(
+      @RequestParam(name = "fur", required = false, defaultValue = "Gray") String fur,
+      @RequestParam(name = "age", required = false, defaultValue = "Adult") String age,
+      @RequestParam(name = "location", required = false, defaultValue = "Above Ground")
+          String location,
+      Model model) {
+    logger.info("Request received for fur: {}, age: {}, location: {}", fur, age, location);
 
-    // TODO: get real values from dataset.
-    model.addAttribute("squirrel_count", "5");
-    List<String> dataPoints = Arrays.asList("1", "2", "3", "4", "5");
-    model.addAttribute("data_points", dataPoints);
+    // Default values (which would be rendered as "No data available.").
+    model.addAttribute("squirrel_count", 0);
+    model.addAttribute("data_points", Arrays.asList(0, 0, 0, 0, 0));
+
+    SquirrelSegment squirrelSegment = retrieveData(fur, age, location);
+    if (squirrelSegment == null) {
+      return "index";
+    }
 
+    model.addAttribute("squirrel_count", squirrelSegment._counter);
+    List<Integer> dataPoints =
+        Arrays.asList(
+            squirrelSegment.Chasing,
+            squirrelSegment.Climbing,
+            squirrelSegment.Eating,
+            squirrelSegment.Foraging,
+            squirrelSegment.Running);
+    model.addAttribute("data_points", dataPoints);
     return "index";
   }
+
+  /**
+   * @return A SquirrelSegment object representing the contents of the JSON file. Returns null if
+   *     we're unable to find the file inside the Cloud Storage bucket.
+   */
+  public SquirrelSegment retrieveData(String fur, String age, String location) {
+    String filePath = fur + "/" + age + "/" + location + "/data.json";
```

Comment:
nit: High risk of malforming the file name. Suggest to utilize `java.nio.file.Paths` class.